### PR TITLE
Prevent cfg simplification from transforming stores of internal pointers

### PR DESCRIPTION
**Pull request by rpshukla**
Transforming stores of internal pointers can cause the gc map to be incorrect.

Signed-off-by: Ryan Shukla <ryans@ibm.com>

### CODE REVIEW

File: compiler/optimizer/OMRCFGSimplifier.cpp [hunk 1]
```diff
@@ -1166,6 +1172,8 @@ bool OMR::CFGSimplifier::simplifyBooleanStore(bool needToDuplicateTree)
       }
    TR::Node *value1 = store1->getChild(valueIndex);
    TR::Node *value2 = store2->getChild(valueIndex);
+   if (value1->isInternalPointer() || value2->isInternalPointer())
```
Comment (andrewcraik) [inline compiler/optimizer/OMRCFGSimplifier.cpp:1175]: Could this moved up with the opcode checks etc and/or adding some tracing to know we passed this test?

**Comment by andrewcraik** [comment]
Apart from one change for tracing this looks good - will approve once updated.

**Comment by rpshukla** [comment]
Moved the checks for internal pointers sooner and added tracing.

**Comment by andrewcraik** [comment]
@genie-omr build all

**Comment by andrewcraik** [comment]
@genie-omr build osx

**Comment by andrewcraik** [comment]
@genie-omr build osx

**Comment by andrewcraik** [comment]
OSX looks to be an existing issue - the failure is not related to this change. Merging.

**Pull request merged**